src/main/scala/ch/epfl/scala/bsp/Bsp.scala

```diff
@@ -0,0 +1,401 @@
+package ch.epfl.scala.bsp
+
+import java.net.{URI, URISyntaxException}
+
+import io.circe.Decoder.Result
+import io.circe.{Decoder, DecodingFailure, HCursor, Json, RootEncoder}
+import io.circe.generic.JsonCodec
+
+final case class Uri private[Uri](val s: String)
```

Comment:
s/[Uri]// ?

I think the constructor is still accessible.

Also, what does `s` stand for in `val s: String`? I would expect `value: String`

Reply:
The constructor is only accessible inside the class and companion `Uri`. I'm changing the name `s` to `value`.